### PR TITLE
Feature/config redux

### PR DESCRIPTION
**Pull request by vinhlee95**

### CODE REVIEW

File: src/configStore.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+// Config redux devtool
+// Redux does not provide types for __REDUX_DEVTOOLS_EXTENSION_COMPOSE__
+declare global {
+	interface Window {
+		__REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose
+	}
+}
+
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
```
Comment (HuyAms) [inline src/configStore.ts]: turn Redux Devtool off in production mode

File: src/configStore.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+const store = createStore(rootReducer, composeEnhancers(applyMiddleware(thunk)))
```
Comment (HuyAms) [inline src/configStore.ts]: consider adding `redux-logger` to middleware (only in development env)
https://github.com/LogRocket/redux-logger

```
import {createLogger} from "redux-logger";

 if (process.env.NODE_ENV !== "production") {
    const logger = createLogger();
    middlewares.push(logger);
  }

```

File: src/configStore.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/**
+ * Config redux store
+ *
+ * @author Vinh Le <lethanhvinh95@gmail.com>
+ *
+ */
+
+import { createStore, applyMiddleware, compose } from 'redux'
+import { rootReducer } from './reducer'
+const { createLogger } = require('redux-logger')
```
Comment (vinhlee95) [inline src/configStore.ts]: This lib is weirdly implemented with TS:
https://github.com/LogRocket/redux-logger/issues/89

File: src/configStore.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	(process.env.NODE_ENV !== 'production' &&
+		window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) ||
+	compose
+
+// Config redux-logger in development
+let logger
+if (process.env.NODE_ENV === `development`) {
+	logger = createLogger()
+}
+
```
Comment (HuyAms) [inline src/configStore.ts]: Refactor these codes to:

```
const middlewares = [thunk]

if (process.env.NODE_ENV === 'development') {
 const logger = createdLogger()
 middlewares.push(logger)
}
...
composeEnhancers(applyMiddleware(...middlewares)),
```

File: src/configStore.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+const composeEnhancers =
+	(process.env.NODE_ENV !== 'production' &&
+		window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) ||
+	compose
+
+// Config redux-logger in development
+let logger
+if (process.env.NODE_ENV === `development`) {
```
Comment (HuyAms) [inline src/configStore.ts]: no need to use string template here. `development`-> 'development'

File: src/configStore.ts (hunk 3 above)
Comment (HuyAms) [reply src/configStore.ts]: change "target": "es5" to "target": "esnext" in tsconfig.json

import {createLogger} from "redux-logger"

File: src/configStore.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+// Config redux devtool
+// Redux does not provide types for __REDUX_DEVTOOLS_EXTENSION_COMPOSE__
+declare global {
+	interface Window {
+		__REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose
+	}
+}
+
+const composeEnhancers =
+	(process.env.NODE_ENV !== 'production' &&
```
Comment (HuyAms) [inline src/configStore.ts]: extract this to `const isProduction=...`

**Pull request merged**